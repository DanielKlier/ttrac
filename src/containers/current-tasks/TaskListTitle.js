import * as React from 'react';
import PropTypes from 'prop-types';
import ToggleDisplay from '../../components/ToggleDisplay';

export default class TaskListTitle extends React.Component {
    // noinspection JSUnusedGlobalSymbols
    static propTypes = {
        title        : PropTypes.string,
        onTextChanged: PropTypes.func
    };

    constructor(props) {
        super(props);

        this.state = {
            isEditing: !props.title,
            text     : this.props.title || ''
        };

        this.titleInput = null;
    }

    componentWillReceiveProps(props) {

    }

    render() {

        return (
            <h4 className="task-list-description">
                <ToggleDisplay show={!this.state.isEditing && this.state.text}>
                    <span onClick={() => this.showEditInput()}>
                        {this.state.text}
                    </span>
                </ToggleDisplay>
                <ToggleDisplay show={this.state.isEditing || !this.state.text}>
                    <input type="text"
                           ref={el => {this.titleInput = el;}}
                           value={this.state.text}
                           onBlur={(e) => this.stopEditing(e.target.value)}
                           placeholder="Click to add a title"
                           onChange={
                               (e) => this.textInputChanged(e.target.value)
                           }
                    />
                </ToggleDisplay>
            </h4>
        );
    }

    showEditInput() {
        const input = this.titleInput;
        this.setState({isEditing: true}, () => {
            input.focus();
            input.select();
        });
    }

    stopEditing(value) {
        this.setState({isEditing: false});
        this.props.onTextChanged(value);
    }

    textInputChanged(value) {
        this.setState({text: value});
    }
}