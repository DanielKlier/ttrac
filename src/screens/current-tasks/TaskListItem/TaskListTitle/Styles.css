.task-list-description {
    margin-bottom: 5px;
    white-space: nowrap;
    text-overflow: ellipsis;
    width: 100%;
    overflow: hidden;
}
