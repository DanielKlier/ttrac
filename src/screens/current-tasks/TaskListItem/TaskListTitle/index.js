import * as React from 'react';
import PropTypes from 'prop-types';
import {noop} from 'lodash';
import ToggleDisplay from '../../../../components/ToggleDisplay';
import './Styles.css';

export default class TaskListTitle extends React.Component {
    // noinspection JSUnusedGlobalSymbols
    static propTypes = {
        title: PropTypes.string,
        onTextChanged: PropTypes.func
    };

    // noinspection JSUnusedGlobalSymbols
    static defaultProps = {
        title: '',
        onTextChanged: noop
    };

    constructor(props) {
        super(props);

        this.state = {
            isEditing: !props.title,
            text: this.props.title || ''
        };

        this.titleInput = null;
    }

    render() {

        return (
            <h4 className="task-list-description">
                <ToggleDisplay show={!this.state.isEditing && this.state.text}>
                    <span className="title" onClick={() => this.showEditInput()}>
                        {this.state.text}
                    </span>
                </ToggleDisplay>
                <ToggleDisplay show={this.state.isEditing || !this.state.text}>
                    <input type="text"
                           className="title-input"
                           ref={el => {
                               this.titleInput = el;
                           }}
                           value={this.state.text}
                           onBlur={(e) => this.stopEditing(e.target.value)}
                           placeholder="Click to add a title"
                           onChange={
                               (e) => this.textInputChanged(e.target.value)
                           }
                    />
                </ToggleDisplay>
            </h4>
        );
    }

    showEditInput() {
        const input = this.titleInput;
        this.setState({isEditing: true}, () => {
            input.focus();
            input.select();
        });
    }

    stopEditing() {
        this.setState({isEditing: false}, () => {
            this.props.onTextChanged(this.state.text);
        });
    }

    textInputChanged(value) {
        this.setState({text: value});
    }
}
