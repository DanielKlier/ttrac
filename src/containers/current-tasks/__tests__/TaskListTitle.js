import React from 'react';
import renderer from 'react-test-renderer';
import TaskListTitle from '../TaskListTitle';

test('TaskListTitle renders correctly', () => {
    expect(renderer.create(
        <TaskListTitle/>
    ).toJSON()).toMatchSnapshot();

    expect(renderer.create(
        <TaskListTitle title="My Title"/>
    ).toJSON()).toMatchSnapshot();
});
