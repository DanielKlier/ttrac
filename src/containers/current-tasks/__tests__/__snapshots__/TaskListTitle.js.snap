// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaskListTitle behaves correctly 1`] = `
<TaskListTitle
  onTextChanged={[Function]}
>
  <h4
    className="task-list-description"
  >
    <toggleDisplay
      show={false}
    >
      <span
        className="hidden"
      >
        <span
          className="title"
          onClick={[Function]}
        />
      </span>
    </toggleDisplay>
    <toggleDisplay
      show={true}
    >
      <span
        className=""
      >
        <input
          className="title-input"
          onBlur={[Function]}
          onChange={[Function]}
          placeholder="Click to add a title"
          type="text"
          value=""
        />
      </span>
    </toggleDisplay>
  </h4>
</TaskListTitle>
`;

exports[`TaskListTitle behaves correctly 2`] = `
<TaskListTitle
  onTextChanged={[Function]}
>
  <h4
    className="task-list-description"
  >
    <toggleDisplay
      show=""
    >
      <span
        className="hidden"
      >
        <span
          className="title"
          onClick={[Function]}
        />
      </span>
    </toggleDisplay>
    <toggleDisplay
      show={true}
    >
      <span
        className=""
      >
        <input
          className="title-input"
          onBlur={[Function]}
          onChange={[Function]}
          placeholder="Click to add a title"
          type="text"
          value=""
        />
      </span>
    </toggleDisplay>
  </h4>
</TaskListTitle>
`;

exports[`TaskListTitle renders correctly 1`] = `
<h4
  className="task-list-description"
>
  <span
    className="hidden"
  >
    <span
      className="title"
      onClick={[Function]}
    >
      
    </span>
  </span>
  <span
    className=""
  >
    <input
      className="title-input"
      onBlur={[Function]}
      onChange={[Function]}
      placeholder="Click to add a title"
      type="text"
      value=""
    />
  </span>
</h4>
`;

exports[`TaskListTitle renders correctly 2`] = `
<h4
  className="task-list-description"
>
  <span
    className=""
  >
    <span
      className="title"
      onClick={[Function]}
    >
      My Title
    </span>
  </span>
  <span
    className="hidden"
  >
    <input
      className="title-input"
      onBlur={[Function]}
      onChange={[Function]}
      placeholder="Click to add a title"
      type="text"
      value="My Title"
    />
  </span>
</h4>
`;
